ui/src/app/pages/admin/admin-workspace.tsx

```diff
@@ -92,35 +116,76 @@ const FileDetailsTable = (props: {data: Array<FileDetail>, bucket: string}) => {
     }
   };
 
-  const formattedData: Array<TableEntry> = data.map(file => {
-    return {
-      name: file.name,
-      size: formatMB(file.sizeInBytes),
-      location: getLocation(file)
+  const nameCell = (file: FileDetail): React.ReactFragment => {
+    const filename = file.name.trim();
+    const filenameText = <span>{filename}</span>;
+    const navigateToPreview = () => {
+      navigate(['admin', 'workspaces', workspaceNamespace, filename],
+          { queryParams: { accessReason: accessReason } });
     };
-  });
 
-  const sortedData = formattedData.sort((a, b) => {
-    const locCmp = a.location.localeCompare(b.location);
-    if (locCmp === 0) {
-      return a.name.localeCompare(b.name);
+    // remove first check after RW-5626
+    if (NOTEBOOKS_DIRECTORY === getLocation(file) && filename.endsWith(NOTEBOOKS_SUFFIX)) {
+      if (file.sizeInBytes > MAX_NOTEBOOK_READ_SIZE_BYTES) {
+        return <FlexRow>
+          {filenameText}
+          <TooltipTrigger
+            content={`Files larger than ${formatMB(MAX_NOTEBOOK_READ_SIZE_BYTES)} MB are too large to preview`}
+          ><Button style={styles.previewButton} disabled={true}>Preview</Button>
+          </TooltipTrigger>
+        </FlexRow>;
+      } else {
+        return <FlexRow>
+          {filenameText}
+          <TooltipTrigger content='Please enter an access reason below' disabled={accessReason && accessReason.trim()}>
+            <Button style={styles.previewButton} disabled={!accessReason || !accessReason.trim()}
+                    onClick={navigateToPreview}>Preview</Button>
+          </TooltipTrigger>
+        </FlexRow>;
+      }
     } else {
-      return locCmp;
+      return filenameText;
     }
-  });
-
-  return <DataTable
-      data-test-id='object-details-table'
-      value={sortedData}
-      style={styles.fileDetailsTable}
-      scrollable={true}
-      paginator={true}
-      paginatorTemplate='CurrentPageReport FirstPageLink PrevPageLink PageLinks NextPageLink LastPageLink RowsPerPageDropdown'
-      currentPageReportTemplate='Showing {first} to {last} of {totalRecords} entries'>
-    <Column field='location' header='Location'/>
-    <Column field='name' header='Filename'/>
-    <Column field='size' header='File size (MB)'/>
-  </DataTable>;
+  };
+
+  const setupTable = () => {
+    setTableData(data
+      .map(file => {
+        return {
+          location: getLocation(file),
+          rawName: file.name,
+          nameCell: nameCell(file),
+          size: formatMB(file.sizeInBytes),
+        }; })
+      .sort((a, b) => {
+        const locCmp = a.location.localeCompare(b.location);
+        if (locCmp === 0) {
+          return a.rawName.localeCompare(b.rawName);
+        } else {
+          return locCmp;
+        }}));
+  };
+
+  useEffect(() => {
```

Comment:
I am not seeing the connection between accessReason and setting the table data - does the table data change as the accessReason changes? If so could this be added to the onChange handler rather than as an effect?

Reply:
The enabled/disabled state of the button changes depending on accessReason.  This way of linking them does feel clunky.

I'll see what I can do with onChange.

Reply:
now updated via onChange